client/Client.cpp

```diff
@@ -399,20 +423,139 @@ namespace client {
       return;
     }
     try {
+      uint32_t currentSeq = _sequence_number.load(std::memory_order_acquire);
       PlayerShootPacket packet = PacketBuilder::makePlayerShoot(
-          x, y, ProjectileType::PLAYER_BASIC,
-          _sequence_number.load(std::memory_order_acquire));
+          x, y, ProjectileType::PLAYER_BASIC, currentSeq);
       send(packet);
     } catch (const std::exception &e) {
       TraceLog(LOG_ERROR, "[SEND SHOOT] Exception: %s", e.what());
     }
   }
 
   /**
-   * @brief Sends a matchmaking request to the connected server.
+   * @brief Record a sent packet for retransmission tracking keyed by its
+   * sequence number.
+   *
+   * Creates an unacknowledged-packet entry containing the serialized packet
+   * bytes, initializes its resend count to zero, sets its last-sent timestamp
+   * to now, and stores it in the client's unacknowledged packet map using the
+   * provided sequence number.
+   *
+   * @param sequence_number Sequence identifier for the packet used as the map
+   * key.
+   * @param packetData Shared pointer to the serialized packet byte buffer to be
+   * resent if unacknowledged.
+   */
+  void Client::addUnacknowledgedPacket(
+      std::uint32_t sequence_number,
+      std::shared_ptr<std::vector<uint8_t>> packetData) {
+    std::lock_guard<std::mutex> lock(_unacknowledgedPacketsMutex);
+    UnacknowledgedPacket packet;
+    packet.data = packetData;
+    packet.resend_count = 0;
+    packet.last_sent = std::chrono::steady_clock::now();
+    _unacknowledged_packets[sequence_number] = packet;
+  }
+
+  /**
+   * @brief Remove the tracked unacknowledged packet with the given sequence
+   * number.
+   *
+   * Removes the entry for the acknowledged packet from the client's
+   * unacknowledged packet store. If no entry exists for the provided sequence
+   * number, a warning is logged and no change is made.
+   *
+   * @param sequence_number Sequence number of the packet to remove.
+   */
+  void Client::removeAcknowledgedPacket(std::uint32_t sequence_number) {
+    std::lock_guard<std::mutex> lock(_unacknowledgedPacketsMutex);
+    auto it = _unacknowledged_packets.find(sequence_number);
+    if (it != _unacknowledged_packets.end()) {
+      TraceLog(LOG_INFO,
+               "[ACK] Client removing acknowledged packet %u (had %zu unacked "
+               "packets)",
+               sequence_number, _unacknowledged_packets.size());
+      _unacknowledged_packets.erase(it);
+      TraceLog(LOG_INFO,
+               "[ACK] Client now has %zu unacknowledged packets remaining",
+               _unacknowledged_packets.size());
+    } else {
+      TraceLog(LOG_WARNING,
+               "[ACK] Client tried to remove non-existent packet %u",
+               sequence_number);
+    }
```

Comment:
à supprimer (les logs & le `else`)

```suggestion
    }
```